core/persistence-jpa/src/main/java/org/apache/syncope/core/persistence/jpa/dao/AbstractAnySearchDAO.java

```diff
@@ -297,26 +301,37 @@ protected String check(final MemberCond cond) {
     }
 
     protected <T extends Any<?>> List<T> buildResult(final List<Object> raw, final AnyTypeKind kind) {
-        List<T> result = new ArrayList<>();
+        LinkedHashSet<String> orderedAnyKeys = extractSortedAnyKeys(raw);
```

Comment:
Any reason not to use `List` instead? e.g.

```java
List<String> orderedAnyKeys = extractSortedAnyKeys(raw);
```

Reply:
Ok, as SQL returns distinct keys, I'll return list.